utils/python/CIME/XML/env_batch.py

```diff
@@ -249,7 +249,7 @@ def set_job_defaults(self, bjobs, pesize=None):
                 task_count = int(task_count)
             queue = self.select_best_queue(task_count)
             self.set_value("JOB_QUEUE", queue, subgroup=job)
-            walltime = self.get_max_walltime(queue)
+            walltime = self.get_max_walltime(queue) if walltime is None else walltime
```

Comment:
Sorry, `os.getenv('CIME_GLOBAL_WALLTIME')` is what you used above. I was wondering why you wouldn't just get a `KeyError`, but now I see that you used `getenv` rather than the dictionary-type interface.